conans/client/graph/proxy.py

```diff
@@ -62,67 +68,95 @@ def _get_recipe(self, reference, check_updates, update, remotes):
         selected_remote = remotes.selected or cur_remote
 
-        check_updates = check_updates or update
+        if check_updates:
 
-        # TODO: cache2.0 check this for new update flows
-        # requested_different_revision = (ref.revision is not None) and cur_revision != ref.revision
-        # if requested_different_revision:
-        #     if check_updates:
-        #         remote, new_ref = self._download_recipe(layout, ref, output, remotes,
-        #                                                 selected_remote)
-        #         status = RECIPE_DOWNLOADED
-        #         return conanfile_path, status, remote, new_ref
-        #     else:
-        #         raise NotFoundException("The '%s' revision recipe in the local cache doesn't "
-        #                                 "match the requested '%s'."
-        #                                 " Use '--update' to check in the remote."
-        #                                 % (cur_revision, repr(ref)))
-
-        if not check_updates:
-            status = RECIPE_INCACHE
-            return conanfile_path, status, cur_remote, ref
-
-        # Checking updates in the server
-        if not selected_remote:
-            status = RECIPE_NO_REMOTE
-            return conanfile_path, status, None, ref
-
-        try:  # get_recipe_manifest can fail, not in server
-            upstream_manifest, ref = self._remote_manager.get_recipe_manifest(ref, selected_remote)
-        except NotFoundException:
-            status = RECIPE_NOT_IN_REMOTE
-            return conanfile_path, status, selected_remote, ref
-
-        read_manifest = recipe_layout.recipe_manifest()
-        # TODO: cache2.0 check this for 2.0
-        if upstream_manifest != read_manifest:
-            if upstream_manifest.time > read_manifest.time:
-                if update:
-                    DiskRemover().remove_recipe(recipe_layout, output=output)
-                    output.info("Retrieving from remote '%s'..." % selected_remote.name)
-                    self._download_recipe(recipe_layout, ref, output, remotes, selected_remote)
-                    status = RECIPE_UPDATED
-                    return conanfile_path, status, selected_remote, ref
+            remote, latest_rrev, remote_time = self._get_rrev_from_remotes(reference,
+                                                                           remotes.values(),
+                                                                           check_all_servers=True)
+            if latest_rrev:
+                # check if we already have the latest in local cache
+                # TODO: cache2.0 here if we already have a revision in the cache but we add the
+                #  --update argument and we find that same revision in server, we will not
+                #  download anything but we will UPDATE the date of that revision in the
+                #  local cache and WE ARE ALSO UPDATING THE REMOTE
```

Comment:
I don't remember how the timestamp in the sever was assigned but... wasn't it automatic with the server time at the time you upload a revision? How could we force a timestamp in the server based on a local one? I'm clearly missing something.

Reply:
Sorry, the sentence is a little bit confusing, I'm referring to update the remote associated to the reference in the database (what previously was stored in the metadata). In this PR only local timestamps are changes because as you say we can't modify the remote ones.